packages/web/app/src/components/ui/changelog/changelog.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { ReactElement, useState } from 'react';
+import { Button } from '@/components/ui/button';
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/ui/popover';
+import { useLocalStorage, useToggle } from '@/lib/hooks';
+import { cn } from '@/lib/utils';
+import { HiveLogo } from '../../v2/icon';
+import { latestChangelogs } from './generated-changelogs';
+
+export type Changelog = {
+  title: string;
+  description: string;
+  userReadItem: boolean;
+  href?: string;
+  route?: string;
+  date: string;
+};
+
+export function Changelog(): ReactElement {
+  const [isOpen, toggle] = useToggle();
+  return <ChangelogPopover isOpen={isOpen} onClose={toggle} />;
+}
+
+function ChangelogPopover({ isOpen, onClose }: { isOpen: boolean; onClose: () => void }) {
+  const [changelogItems, setChangelogItems] = useLocalStorage<Changelog[]>(
+    'changelogItems',
+    latestChangelogs,
+  );
```

Comment:
i manually edited the generated changelog file

![image](https://github.com/kamilkisiela/graphql-hive/assets/14338007/24c340e3-f89b-4083-af64-1e80bb31c777)

After refreshing the page the changes did not show up:

![image](https://github.com/kamilkisiela/graphql-hive/assets/14338007/fe3ac681-6eeb-4ac4-b2ef-a8b50447ed80)


I suspect this is because the items are always read from local storage.

____

I don't think we need to store the full list in local storage, we can always use the changelog items from the generated file.

Then we only store the d/url of the first item in the changelog items in the local storage.

If the user opens the menu, we set the local storage value to the first item in the list

if the local storage value is not equal to the first item in the generated list, we show a red notification dot

![Screenshot 2024-03-14 at 13 25 31](https://github.com/kamilkisiela/graphql-hive/assets/14338007/8e48b498-701e-438d-be1b-3807aa768206)